nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/mysql/event/io/InsertRowsWriter.java

```diff
@@ -0,0 +1,90 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard.db.impl.mysql.event.io;
+
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processors.standard.GetChangeDataCaptureMySQL;
+import org.apache.nifi.processors.standard.db.event.ColumnDefinition;
+import org.apache.nifi.processors.standard.db.impl.mysql.event.InsertRowsEventInfo;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.BitSet;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.apache.nifi.processors.standard.db.impl.mysql.MySQLCDCUtils.getWritableObject;
+
+/**
+ * A writer class to output MySQL binlog "write rows" (aka INSERT) events to flow file(s).
+ */
+public class InsertRowsWriter extends AbstractBinlogTableEventWriter<InsertRowsEventInfo> {
+
+    /**
+     * Creates and transfers a new flow file whose contents are the JSON-serialized value of the specified event, and the sequence ID attribute set
+     *
+     * @param session   A reference to a ProcessSession from which the flow file(s) will be created and transferred
+     * @param eventInfo An event whose value will become the contents of the flow file
+     * @return The next available CDC sequence ID for use by the CDC processor
+     */
+    public long writeEvent(final ProcessSession session, final InsertRowsEventInfo eventInfo, final long currentSequenceId) {
+        final AtomicLong seqId = new AtomicLong(currentSequenceId);
+        for (Serializable[] row : eventInfo.getRows()) {
+
+            FlowFile flowFile = session.create();
+            flowFile = session.write(flowFile, outputStream -> {
+
+                super.startJson(outputStream, eventInfo);
+                super.writeJson(eventInfo);
+
+                final BitSet bitSet = eventInfo.getIncludedColumns();
+                writeRow(eventInfo, row, bitSet);
+
+                super.endJson();
+            });
+
+            flowFile = session.putAllAttributes(flowFile, getCommonAttributes(seqId.get(), eventInfo));
+            session.transfer(flowFile, GetChangeDataCaptureMySQL.REL_SUCCESS);
+            seqId.getAndIncrement();
+        }
+        return seqId.get();
+    }
+
+    protected void writeRow(InsertRowsEventInfo event, Serializable[] row, BitSet includedColumns) throws IOException {
+        jsonGenerator.writeArrayFieldStart("columns");
+        int i = includedColumns.nextSetBit(0);
+        while (i != -1) {
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeNumberField("id", i + 1);
+            ColumnDefinition columnDefinition = event.getColumnByIndex(i);
+            Integer columnType = null;
+            if (columnDefinition != null) {
+                jsonGenerator.writeStringField("name", columnDefinition.getName());
+                columnType = (int) columnDefinition.getType();
```

Comment:
Per the above comment/response, I've changed the getType() method to return an int, I only use Integer in case we don't have a column definition, then we can write "null" as the value for the column type. I'd prefer to use the JDBC Types constants rather than our own enum, as that is what will be used 99% of the time.